src/main/java/org/jabref/gui/mergeLibraries/MergeCommand.java

```diff
@@ -0,0 +1,198 @@
+
+
+        Optional<BibDatabaseContext> database = stateManager.getActiveDatabase();
+        if(path.isPresent() && database.isPresent()) {
+            try {
+                doMerge(path.get(), database.get().getDatabase());
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+    public BibDatabase doMerge(Path path, BibDatabase database) throws IOException {
+        //find all the .lib files by crawling in doMerge and merge them
+        List<BibEntry> toAdd = new ArrayList<>();
+        for(Path f : getAllFiles(path)) {
+            ParserResult result;
+
+            //try to convert the .bib file to a database
+
+            try {
+                result = loadDatabase(f);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+
+
+            /*then go through each entry in the current database and check
+              against the following rules for merging:
+              1. if the entry is equal to any in the main database then ignore the new entry
+              2. if the entry has the same key as any in the main database then ignore the new entry
+              3. if the entry is a duplicate of any in the main database then ignore the new entry
+            */
+            BibEntryTypesManager entryTypesManager = new BibEntryTypesManager();
+
+
+            for (BibEntry entry : result.getDatabase().getEntries()) {
+                boolean addFlag = true;
+                for (BibEntry dbEntry : database.getEntries()) {
+
+                    if (entry.equals(dbEntry)) {
+                        addFlag = false;
+                        break;
+                    }
+
+                    if (entry.getCitationKey().equals(dbEntry.getCitationKey())) {
+                        addFlag = false;
+                        break;
+                    }
+
+                    if (new DuplicateCheck(entryTypesManager).isDuplicate(entry, dbEntry, BibDatabaseMode.BIBTEX)){
+                        addFlag = false;
+                        break;
+                    }
+                }
+                if (addFlag) {
+                    toAdd.add(entry);
+                }
+            }
+        }
+        database.insertEntries(toAdd.stream().toList());
+//
+        //then add the database to the StateManager's activeDatabase
+        return database;
+    }
+
+    /**
+     *  getAllFiles: Recursively crawls through a directory to find all .bib files
+     *
+     * @param path the path to a directory containing the .bib files we are looking
+     *             to merge
+     * @return all the .bib files in a given directory and subdirectories
+     */
+    private List<Path> getAllFiles(Path path) throws IOException {
+        List<Path> paths = new ArrayList<>();
+
+        try (DirectoryStream<Path> stream = Files.newDirectoryStream(path)) {
+            for (Path entry : stream) {
```

Comment:
This code can be replaced using [Files#find](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Files.html#find-java.nio.file.Path-int-java.util.function.BiPredicate-java.nio.file.FileVisitOption...-).